perfkitbenchmarker/linux_benchmarks/specsfs2014_benchmark.py

```diff
@@ -0,0 +1,290 @@
+  mount_dir = 'spec_mnt/'
+  vm.RemoteCommand('mkdir %s' % mount_dir)
+  vm.RemoteCommand('mkdir %s' % _SPEC_DIR)
+  vm.PushFile(data.ResourcePath(_SPEC_SFS_2014_ISO))
+  vm.PushFile(data.ResourcePath(_SPEC_SFS_2014_LICENSE), _SPEC_DIR)
+  vm.RemoteCommand('sudo mount -t iso9660 -o loop %s %s' %
+                   (_SPEC_SFS_2014_ISO, mount_dir))
+  vm.RemoteCommand('cp -r %s/* %s' % (mount_dir, _SPEC_DIR))
+  vm.RemoteCommand('sudo umount {0} && sudo rm -rf {0}'.format(mount_dir))
+  if gluster_server:
+    vm.Install('gluster')
+    gluster.MountGluster(vm, gluster_server, _VOLUME_NAME, _MOUNT_POINT)
+
+
+def _PrepareServer(vm):
+  """Prepares a Gluster server by installing GlusterFS."""
+  vm.Install('gluster')
+
+
+def _ConfigureSpec(prime_client, clients):
+  """Configures SPEC SFS 2014 on the prime client.
+
+  Args:
+    prime_client: The VM from which SPEC will be controlled.
+    clients: A list of SPEC client VMs (including the prime_client).
+  """
+  config_path = posixpath.join(_SPEC_DIR, 'sfs_rc')
+
+  if FLAGS.specsfs2014_config:
+    prime_client.PushFile(
+        data.ResourcePath(FLAGS.specsfs2014_config), config_path)
+  else:
+    stdout, _ = prime_client.RemoteCommand('pwd')
+    exec_path = posixpath.join(stdout.strip(), _SPEC_DIR, 'binaries',
+                               'linux', 'x64', 'netmist')
+    configuration_overrides = {
+        'USER': prime_client.user_name,
+        'EXEC_PATH': exec_path.replace('/', '\/'),
+        'CLIENT_MOUNTPOINTS': _MOUNTPOINTS_FILE,
+        'BENCHMARK': FLAGS.specsfs2014_benchmark,
+        'LOAD': FLAGS.specsfs2014_load,
+        'NUM_RUNS': FLAGS.specsfs2014_num_runs,
+        'INCR_LOAD': FLAGS.specsfs2014_incr_load
+    }
+    # Any special characters in the overrides dictionary should be escaped so
+    # that they don't interfere with sed.
+    sed_expressions = ' '.join(
+        ['-e "s/{0}=.*/{0}={1}/"'.format(k, v)
+         for k, v in configuration_overrides.iteritems()])
+    sed_cmd = 'sudo sed -i {0} {1}'.format(sed_expressions, config_path)
+    prime_client.RemoteCommand(sed_cmd)
+
+  with vm_util.NamedTemporaryFile() as tf:
+    for client in clients:
+      tf.write('%s %s\n' % (client.internal_ip, _MOUNT_POINT))
+    tf.close()
+    prime_client.PushFile(tf.name, posixpath.join(_SPEC_DIR, _MOUNTPOINTS_FILE))
+
+
+def Prepare(benchmark_spec):
+  """Install SPEC SFS 2014.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  gluster_servers = benchmark_spec.vm_groups['gluster_servers']
+  clients = benchmark_spec.vm_groups['clients']
+  prime_client = clients[0]
+
+  vm_util.RunThreaded(_PrepareServer, gluster_servers)
+  gluster.ConfigureServers(gluster_servers, _VOLUME_NAME)
+
+  server = gluster_servers[0] if gluster_servers else None
+  args = [((client, server), {}) for client in clients]
+  vm_util.RunThreaded(_PrepareClient, args)
```

Comment:
It looks like everything above this RunThreaded is about setting up gluster, everything below is SPEC related, and _PrepareClient is a bit mixed. It may be good to split _PrepareClient, but that can happen later.